Sources/HAP/Server/Device.swift

```diff
@@ -124,7 +129,7 @@ public class Device {
             configuration = try decoder.decode(Configuration.self, from: configData)
```

Comment:
So if the setupCode changed from `.automatic` to `.default`, we would effectively be ignoring that here. That might be confusing for users. What would be the best solution here to handle situations like this, also the other way around? Does the pairing / auto-setup break if we do change the `setupCode`?

Reply:
That's why I felt my solution was a bit of a kludge. Pairing won't break, if the code is changed. The hash code and X-HM will modify accordingly. What I could do is make the configuration store "lastSetupCode", and override that with one supplied by a user if specified. Your thoughts ?

Reply:
I have reorganised things a little. If a `.predefined(setupCode)` is given as an argument, it will override whatever is in the configuration. I like this better. I still don't like the SetupCode enum names. I think I would prefer `.automatic` and `.override(setupCode)`, but I've kept your suggestion.

Reply:
I’m fine with override as well!

Reply:
OK, just refactored